### PR TITLE
Fixes analytics generation

### PR DESCRIPTION
**Pull request by mike-lvov**
closes https://github.com/sparkletown/internal-sparkle-issues/issues/1641

I also improved the script
- removed useless files
- you should only pass world slug now

### CODE REVIEW

File: functions/src/analytics.ts [hunk 1]
```diff
@@ -58,20 +61,33 @@ const getUsersWithVisits = async (venueIdsArray: string[]) => {
...
 
   await checkIsAdmin(context.auth.token.user_id);
 
-  const venueIdsArray = Array.isArray(data?.venueIds)
-    ? data.venueIds
-    : (data?.venueIds ?? "").split(",");
+  const matchingVenues = await admin
+    .firestore()
+    .collection("venues")
+    .where("worldId", "==", data.worldSlug)
```
Comment (colinhowe) [inline functions/src/analytics.ts:77]:
```suggestion
    .where("slug", "==", data.worldSlug)
```

File: functions/src/analytics.ts [hunk 2]
```diff
@@ -58,20 +61,33 @@ const getUsersWithVisits = async (venueIdsArray: string[]) => {
...
+  if (!context.auth) {
+    throw new HttpsError("internal", `No authentication context`);
+  }
 
   await checkIsAdmin(context.auth.token.user_id);
 
-  const venueIdsArray = Array.isArray(data?.venueIds)
-    ? data.venueIds
-    : (data?.venueIds ?? "").split(",");
+  const matchingVenues = await admin
```
Comment (colinhowe) [inline functions/src/analytics.ts:74]:
```suggestion
  const matchingSpaces = await admin
```

File: functions/src/analytics.ts [hunk 3]
```diff
@@ -58,20 +61,33 @@ const getUsersWithVisits = async (venueIdsArray: string[]) => {
...
+    .get();
+
+  if (matchingVenues.empty) {
+    throw new HttpsError(
+      "internal",
+      `The world ${data.worldSlug} does not have any venues`
+    );
+  }
+
+  const venueIds = matchingVenues.docs.map((venue) => venue.id);
```
Comment (colinhowe) [inline functions/src/analytics.ts:87]:
```suggestion
  const spaceIds = matchingVenues.docs.map((venue) => venue.id);
```

File: functions/src/analytics.ts [hunk 4]
```diff
@@ -93,7 +109,7 @@ const generateAnalytics: HttpsFunctionHandler<{
     {}
   );
 
-  const allSpaceVisitsFileName = "allSpaceVisits.csv";
+  // const allSpaceVisitsFileName = "allSpaceVisits.csv";
```
Comment (colinhowe) [inline functions/src/analytics.ts:112]: Delete?

**Review by colinhowe — CHANGES_REQUESTED** [review]
Big change to `package-lock` intentional?

File: functions/src/analytics.ts (hunk 1 above)
Comment (mike-lvov) [reply functions/src/analytics.ts:77]: changing worldId to slug doesn't make sense. I want to filter venues/spaces by worldId

File: functions/src/analytics.ts (hunk 2 above)
Comment (mike-lvov) [reply functions/src/analytics.ts]: Renamed in https://github.com/sparkletown/sparkle/pull/2826/commits/0633e23d3105f387087ff095df686fb6e7ce6139

File: functions/src/analytics.ts (hunk 3 above)
Comment (mike-lvov) [reply functions/src/analytics.ts]: Renamed in https://github.com/sparkletown/sparkle/pull/2826/commits/0633e23d3105f387087ff095df686fb6e7ce6139

File: functions/src/analytics.ts (hunk 4 above)
Comment (mike-lvov) [reply functions/src/analytics.ts]: Deleted in https://github.com/sparkletown/sparkle/pull/2826/commits/0633e23d3105f387087ff095df686fb6e7ce6139

**Comment by mike-lvov** [comment]
Will remove both package-lock.json files

**Review by colinhowe — APPROVED** [review]

**Comment by goran-peoski-work** [comment]
> Will remove both package-lock.json files

This is how I deal with those:
1. Merge from staging
2. Delete `package-lock.json`s
3. Finish the merge
4. Do `npm run npm:install` to regenerate them (or `npm run npm:generate-lockfile`)
5. Do `git commit -a --ammend --no-edit` to add the missing lock files
6. PROFIT (usually)

**Pull request merged**